{"metadata":{"fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"ByteToMessageDecoderError"}],"roleHeading":"Enumeration","title":"ByteToMessageDecoderError","role":"symbol","symbolKind":"enum","modules":[{"name":"NIOCore"}],"externalID":"s:7NIOCore25ByteToMessageDecoderErrorO","navigatorTitle":[{"kind":"identifier","text":"ByteToMessageDecoderError"}]},"abstract":[{"type":"text","text":"Common errors thrown by "},{"type":"codeVoice","code":"ByteToMessageDecoder"},{"type":"text","text":"s."}],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError"},"primaryContentSections":[{"declarations":[{"platforms":["Linux"],"tokens":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"ByteToMessageDecoderError"}],"languages":["swift"]}],"kind":"declarations"}],"sections":[],"topicSections":[{"identifiers":["doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError\/PayloadTooLargeError"],"title":"Structures"},{"identifiers":["doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError\/dataReceivedInErrorState(_:_:)","doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError\/leftoverDataWhenDone(_:)"],"title":"Enumeration Cases"}],"schemaVersion":{"patch":0,"minor":3,"major":0},"kind":"symbol","hierarchy":{"paths":[["doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore"]]},"variants":[{"paths":["\/documentation\/niocore\/bytetomessagedecodererror"],"traits":[{"interfaceLanguage":"swift"}]}],"relationshipsSections":[{"identifiers":["doc:\/\/co.pointfree.VaporRouting\/s5ErrorP","doc:\/\/co.pointfree.VaporRouting\/s8SendableP"],"title":"Conforms To","type":"conformsTo","kind":"relationships"}],"references":{"doc://co.pointfree.VaporRouting/documentation/NIOCore/ByteToMessageDecoderError/PayloadTooLargeError":{"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError\/PayloadTooLargeError","abstract":[{"type":"text","text":"This error can be thrown by "},{"code":"ByteToMessageDecoder","type":"codeVoice"},{"type":"text","text":"s if the incoming payload is larger than the max specified."}],"title":"ByteToMessageDecoderError.PayloadTooLargeError","role":"symbol","url":"\/documentation\/niocore\/bytetomessagedecodererror\/payloadtoolargeerror","kind":"symbol","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"PayloadTooLargeError"}],"type":"topic","navigatorTitle":[{"kind":"identifier","text":"PayloadTooLargeError"}]},"doc://co.pointfree.VaporRouting/documentation/NIOCore":{"kind":"symbol","url":"\/documentation\/niocore","title":"NIOCore","role":"collection","type":"topic","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore","abstract":[]},"doc://co.pointfree.VaporRouting/documentation/NIOCore/ByteToMessageDecoderError/dataReceivedInErrorState(_:_:)":{"url":"\/documentation\/niocore\/bytetomessagedecodererror\/datareceivedinerrorstate(_:_:)","type":"topic","abstract":[{"text":"More data has been received by a ","type":"text"},{"type":"codeVoice","code":"ByteToMessageHandler"},{"type":"text","text":" despite the fact that an error has previously been"},{"type":"text","text":" "},{"type":"text","text":"emitted. The associated "},{"code":"Error","type":"codeVoice"},{"type":"text","text":" is the error previously emitted and the "},{"code":"ByteBuffer","type":"codeVoice"},{"type":"text","text":" is the extra data that has"},{"type":"text","text":" "},{"type":"text","text":"been received. The common cause for this error to be emitted is the user not having torn down the "},{"code":"Channel","type":"codeVoice"},{"type":"text","text":" "},{"type":"text","text":"after previously an "},{"code":"Error","type":"codeVoice"},{"type":"text","text":" has been sent through the pipeline using "},{"code":"fireErrorCaught","type":"codeVoice"},{"type":"text","text":"."}],"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError\/dataReceivedInErrorState(_:_:)","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"dataReceivedInErrorState"},{"kind":"text","text":"("},{"kind":"typeIdentifier","preciseIdentifier":"s:s5ErrorP","text":"Error"},{"kind":"text","text":", "},{"kind":"typeIdentifier","preciseIdentifier":"s:7NIOCore10ByteBufferV","text":"ByteBuffer"},{"kind":"text","text":")"}],"title":"ByteToMessageDecoderError.dataReceivedInErrorState(_:_:)","kind":"symbol","role":"symbol"},"doc://co.pointfree.VaporRouting/s5ErrorP":{"identifier":"doc:\/\/co.pointfree.VaporRouting\/s5ErrorP","type":"unresolvable","title":"Swift.Error"},"doc://co.pointfree.VaporRouting/s8SendableP":{"identifier":"doc:\/\/co.pointfree.VaporRouting\/s8SendableP","type":"unresolvable","title":"Swift.Sendable"},"doc://co.pointfree.VaporRouting/documentation/NIOCore/ByteToMessageDecoderError":{"url":"\/documentation\/niocore\/bytetomessagedecodererror","type":"topic","abstract":[{"text":"Common errors thrown by ","type":"text"},{"type":"codeVoice","code":"ByteToMessageDecoder"},{"type":"text","text":"s."}],"navigatorTitle":[{"text":"ByteToMessageDecoderError","kind":"identifier"}],"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"ByteToMessageDecoderError","kind":"identifier"}],"title":"ByteToMessageDecoderError","kind":"symbol","role":"symbol"},"doc://co.pointfree.VaporRouting/documentation/NIOCore/ByteToMessageDecoderError/leftoverDataWhenDone(_:)":{"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/ByteToMessageDecoderError\/leftoverDataWhenDone(_:)","url":"\/documentation\/niocore\/bytetomessagedecodererror\/leftoverdatawhendone(_:)","kind":"symbol","type":"topic","abstract":[{"text":"This error can be thrown by ","type":"text"},{"code":"ByteToMessageDecoder","type":"codeVoice"},{"text":"s if there was unexpectedly some left-over data when the","type":"text"},{"text":" ","type":"text"},{"code":"ByteToMessageDecoder","type":"codeVoice"},{"text":" was removed from the pipeline or the ","type":"text"},{"code":"Channel","type":"codeVoice"},{"text":" was closed.","type":"text"}],"title":"ByteToMessageDecoderError.leftoverDataWhenDone(_:)","role":"symbol","fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"leftoverDataWhenDone","kind":"identifier"},{"text":"(","kind":"text"},{"text":"ByteBuffer","preciseIdentifier":"s:7NIOCore10ByteBufferV","kind":"typeIdentifier"},{"text":")","kind":"text"}]}}}